src/mlpack/core/util/arma_traits.hpp

```diff
@@ -298,6 +298,54 @@ struct IsBaseMatType<arma::SpRow<eT>>
   constexpr static bool value = true;
 };
 
+template<typename MatType>
+struct IsSparseMat
+{
+  constexpr static bool value = false;
+};
+
+template<typename eT>
+struct IsSparseMat<arma::SpMat<eT>>
+{
+  constexpr static bool value = true;
+};
+
+template<typename MatType>
+struct IsCol
+{
+  constexpr static bool value = false;
+};
+
+template<typename eT>
+struct IsCol<arma::Col<eT>>
+{
+  constexpr static bool value = true;
+};
```

Comment:
Technically all of these need to be defined for `const arma::Col<eT>` too, etc.  We can handle that in another PR---I see that everything in `arma_traits.hpp` needs to be fixed in this way.